cmd/microcloud/main_init.go

```diff
@@ -491,6 +503,22 @@ func AddPeers(sh *service.Handler, systems map[string]InitSystem) error {
 		}
 	}
 
+	// After all systems have joined and that the database has been updated and
+	// if some nodes had been configured with custom encapsulation IPs, we can check the members' API extensions
+	// If they don't have the `custom_encapsulation_ip` extension, output that the supplied custom encapsulation IPs
+	// were ignored.
+	if len(membersWithCustomEncapsulationIP) > 0 {
+		hasFeature, err := sh.Services[types.MicroOVN].(*service.OVNService).HasCustomEncapsulationIpFeature(context.Background())
+		if err != nil {
+			return err
+		}
+
+		if !hasFeature {
+			fmt.Println("Underlay network setup for MicroOVN was not applied because your MicroOVN memmbers do not support custom encapsulation IP for the Geneve tunnel, skipping")
```

Comment:
Why are we checking this after joining nodes together? The list of API extensions is stored in memory during on first star, so why can't we check this earlier?

Reply:
Ah yes, true. I faced the issue of the DB not initialized at first (for the joining members), that's why I put the check at the end. But I didn't remember that it was also in memory.... let's do that then. Thanks!

Reply:
@masnax Then, I'm wondering, do we need this function for checking the database if we can check it in memory before?

Reply:
Maybe what would work best here is extending the `Service` interface with a new `SupportsFeature` function that checks the in-memory API extensions of the other snaps.


Since we're going to introduce enforced versioning of the snaps, I think for the purposes of this PR, it's safe to assume that if your local MicroOVN supports the feature, and you were actually able to find systems that are eligible to cluster with you, then you don't have to check those systems since they must already be at the same version as you.

Right now that validation isn't in place but it will be before the LTS so I think it's fine here to just check the local snaps.

Reply:
We will need this then https://github.com/canonical/microcluster/pull/168

Reply:
@masnax updated :)